crypto/src/transaction.rs

```diff
@@ -173,6 +174,36 @@ impl Transaction {
             .verify(&transaction_body_serialized, &self.binding_sig)
             .is_ok()
     }
+
+    /// Verify the binding signature with the given value balance.
+    ///
+    /// This is used for transactions that have a non-zero value balance, such as
+    /// the genesis transaction.
+    pub fn verify_binding_sig_with_value_balance(&self, value_balance: decaf377::Element) -> bool {
+        let mut value_commitments = decaf377::Element::default();
+        for action in self.transaction_body.actions.clone() {
+            match action {
+                Action::Output(inner) => {
+                    value_commitments -= inner.body.value_commitment.0;
+                }
+                Action::Spend(inner) => {
+                    value_commitments += inner.body.value_commitment.0;
+                }
+            }
+        }
+
+        // For the genesis transaction there will be a non-zero value balance since we are creating value.
+        let binding_verification_key_bytes: VerificationKeyBytes<Binding> =
+            (value_commitments - value_balance).compress().0.into();
+        let binding_verification_key: VerificationKey<Binding> = binding_verification_key_bytes
+            .try_into()
+            .expect("verification key is valid");
+
+        let transaction_body_serialized: Vec<u8> = self.transaction_body.clone().into();
+        binding_verification_key
+            .verify(&transaction_body_serialized, &self.binding_sig)
+            .is_ok()
+    }
```

Comment:
Do we still need this function, if we're not going to be verifying the signatures on the genesis transactions, where it would be used? It doesn't look like it's in use anywhere as yet.

Reply:
yeah this would be something we'd use if we have any other transactions with non-zero value balance, but I'll snip it out for now